mobile/src/main/java/ch/epfl/sdp/mobile/ui/profile/ProfileScreen.kt

```diff
@@ -53,6 +57,8 @@ fun <C : ChessMatch> ProfileScreen(
       onMatchClick = state::onMatchClick,
       lazyColumnState = lazyColumnState,
       modifier = modifier)
+
+  BackButton(onClick = state::onBackClick, modifier = modifier.offset(16.dp, 16.dp))

```

Comment:
You should not add `BackButton` here. Instead, you should edit the `ProfileHeader` to give it (an optional) slot which could be used to display a back action.

In general, it is a bad practice to have multiple top-level composables in a function without a receiver. Indeed, the `modifier` argument should not be passed to all children, and the composable will behave differently if it's displayed in another layout (typically, calling this function within a `Row` will place your composables horizontally, and calling it in a `Column` will place them on top of each other).

Reply:
Oh I see, thank you

Reply:
Actually, after a second scan of the code I don't really understand why we would have to pass the `BackButton` composable as a parameter to the `ProfileHeader` why not call it inside directly like I do in [`61d72`](https://github.com/epfl-SDP/android/pull/303/commits/61d7230d9920b7bdc7ec9171023618f9edae07f7)